pkg/apis/admission/types.go

```diff
@@ -0,0 +1,160 @@
+	metav1.TypeMeta
+
+	// ResourceInitializers is a list of resources and their default initializers
+	ResourceInitializers []ResourceDefaultInitializer
+
+	// ExternalAdmissionHooks is a list of external admission webhooks and the
+	// affected resources and operations.
+	ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// ResourceDefaultInitializer describes the default initializers that will be
+// applied to a resource. The order of initializers is sensitive.
+type ResourceDefaultInitializer struct {
+	// APIGroup is the API group of the resource
+	APIGroup string
+
+	// APIVersions is the API Versions of the resource
+	// '*' means all API Versions.
+	// If '*' is present, the length of the slice must be one.
+	// Defaults to '*'.
+	APIVersions []string
+
+	// Resource is resource to be initialized
+	Resource string
+
+	// Initializers is a list of initializers that will be applied to the
+	// resource by default. It is order-sensitive.
+	Initializers []Initializer
+}
+
+// Initializer describes the name and the failure policy of an initializer.
+type Initializer struct {
+	// Name is the identifier of the initializer. It will be added to the
+	// object that needs to be initialized.
+	Name string
+
+	// FailurePolicy defines what happens if there is an initializer controller
+	// fails to takes action. Allowed values are Ignore, or Fail. If "Ignore" is
+	// set, initializer is removed from the initializers list of an object if
+	// the timeout is reached; If "Fail" is set, apiserver returns timeout error
+	// if the timeout is reached.
+	FailurePolicy *FailurePolicyType
+}
+
+type FailurePolicyType string
+
+const (
+	// Ignore means the initilizer is removed from the initializers list of an
+	// object if the initializer is timed out.
+	Ignore FailurePolicyType = "Ignore"
+	// For 1.7, only "Ignore" is allowed. "Fail" will be allow when the
+	// extensible admission feature is more mature.
+	Fail FailurePolicyType = "Fail"
+)
+
+// ExternalAdmissionHook describes an external admission webhook and the
+// resources and operations it applies to.
+type ExternalAdmissionHook struct {
+	// The name of the external admission webhook.
+	Name string
+
+	// ClientConfig defines how to communicate with the hook.
+	ClientConfig AdmissionHookClientConfig
+
+	// Rules describes what operations on what resources/subresources the webhook cares about.
+	// The webhook cares about an operation if it matches _any_ Rule.
+	Rules []Rule
+
+	// FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+	// allowed values are Ignore or Fail. Defaults to Ignore.
+	FailurePolicy FailurePolicyType
+}
+
+// Rule describes the Verbs and Resources an admission hook cares about. Each
+// Rule is a tuple of Verbs and Resources.It is recommended to make sure all
+// the tuple expansions are valid.
+type Rule struct {
+	// Verbs is the verbs the admission hook cares about - CREATE, UPDATE, or *
+	// for all verbs.
+	// Defaults to '*'.
```

Comment:
It's to prevent accident, signing up for all Verbs/Resources should be an explicit decision rather than the default. Also it's consistent with rbac.

Could you point me to the exact comment in #34348? The thread is too long :)